tools/idf_monitor.py

```diff
@@ -130,12 +131,29 @@ class ConsoleReader(StoppableThread):
     """ Read input keys from the console and push them to the queue,
     until stopped.
     """
-    def __init__(self, console, event_queue):
+    def __init__(self, console, event_queue, running_on_wsl=False):
         super(ConsoleReader, self).__init__()
         self.console = console
         self.event_queue = event_queue
+        self.tid = None
+        self.pthread_self = None
+        self.pthread_kill = None
+        if running_on_wsl:
+            libpthread = ctypes.CDLL("libpthread.so.0")
+            self.pthread_self = libpthread.pthread_self
+            self.pthread_self.argtypes = [ ]
+            self.pthread_self.restype = ctypes.c_void_p
+            self.pthread_kill = libpthread.pthread_kill
+            self.pthread_kill.argtypes = [ ctypes.c_void_p, ctypes.c_int ]
+            self.pthread_kill.restype = ctypes.c_int
+            signal.signal(signal.SIGRTMIN, self._handler)
+
+    def _handler(self, signum, frame):
+        pass
 
     def run(self):
+        if self.pthread_self != None:
```

Comment:
Typical Python style is to use "is not None" instead of "!= None".

In this case (and in the same check below) I'd be more comfortable using `if running_on_wsl` everywhere as this makes it easier to trace all the of the WSL workarounds (which can hopefully some day be removed!)

Reply:
(Also if you don't need to test whether `self.pthread_self` is None, you can remove the `self.pthread_self = None` default value settings from the constructor)